xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererStarfish.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "RendererStarfish.h"
+
+#include "../RenderFactory.h"
+#include "DVDCodecs/Video/DVDVideoCodecStarfish.h"
+#include "ServiceBroker.h"
+#include "rendering/gles/RenderSystemGLES.h"
+#include "settings/MediaSettings.h"
+#include "utils/log.h"
+#include "windowing/wayland/WinSystemWaylandWebOS.h"
+
+CRendererStarfish::CRendererStarfish()
+{
+  CLog::Log(LOGINFO, "Instancing CRendererStarfish");
+}
+
+CRendererStarfish::~CRendererStarfish()
+{
+}
+
+CBaseRenderer* CRendererStarfish::Create(CVideoBuffer* buffer)
+{
+  if (buffer && dynamic_cast<CStarfishVideoBuffer*>(buffer))
+    return new CRendererStarfish();
+  return nullptr;
+}
+
+bool CRendererStarfish::Configure(const VideoPicture& picture, float fps, unsigned int orientation)
+{
+  m_iFlags = GetFlagsChromaPosition(picture.chroma_position) |
+             GetFlagsColorMatrix(picture.color_space, picture.iWidth, picture.iHeight) |
+             GetFlagsColorPrimaries(picture.color_primaries) |
+             GetFlagsStereoMode(picture.stereoMode);
+
+  return CLinuxRendererGLES::Configure(picture, fps, orientation);
+}
+
+bool CRendererStarfish::Register()
+{
+  VIDEOPLAYER::CRendererFactory::RegisterRenderer("starfish", CRendererStarfish::Create);
+  return true;
+}
+
+void CRendererStarfish::ManageRenderArea()
+{
+  CBaseRenderer::ManageRenderArea();
+
+  if ((m_exportedDestRect != m_destRect || m_exportedSourceRect != m_sourceRect) &&
+      !m_sourceRect.IsEmpty() && !m_destRect.IsEmpty())
+  {
+    auto origRect =
+        CRect{0, 0, static_cast<float>(m_sourceWidth), static_cast<float>(m_sourceHeight)};
+    dynamic_cast<KODI::WINDOWING::WAYLAND::CWinSystemWaylandWebOS*>(CServiceBroker::GetWinSystem())
```

Comment:
```suggestion
    static_cast<KODI::WINDOWING::WAYLAND::CWinSystemWaylandWebOS*>(CServiceBroker::GetWinSystem())
```

Reply:
Fixed

Reply:
there are some convos like this were you marked them as resolved, but the code is not fixed - lost in rebase?

Reply:
Not sure what happened but I went over the requested changes again and applied them again